From afd4afde4e4a39999952ba6b112eab2dd2e829e4 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Thu, 11 May 2023 19:51:09 +0200
Subject: [PATCH 0752/1695] plat-stm32mp1: stm32mp1_pwr: add dsb in pwr driver

Add missing barrier in the driver for access to register,
mapped as device memory (shareable, bufferable).

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 813faccaadc8c3bd864672fdfcbda18997da073b plat-stm32mp1: stm32mp1_pwr: add dsb in pwr driver]
Change-Id: Ic7af9a2c3fe72ee6c25d009aca3a82df08c716dc
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/305824
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/373110
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pwr.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pwr.c b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pwr.c
index 2ecfd9853..dcc0827fd 100644
--- a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pwr.c
+++ b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pwr.c
@@ -92,6 +92,8 @@ void stm32mp1_pwr_regulator_set_state(enum pwr_regulator id, bool enable)
 		if (!(io_read32(cr3) & ready_mask))
 			panic();
 	} else {
+		/* Make sure the previous operations are visible */
+		dsb();
 		io_clrbits32(cr3, enable_mask);
 	}
 }
-- 
2.47.2

